opencog/atoms/execution/Instantiator.cc

```diff
@@ -275,7 +275,10 @@ Handle Instantiator::walk_tree(const Handle& expr)
 	HandleSeq oset_results;
 	bool changed = walk_tree(oset_results, lexpr->getOutgoingSet());
 	if (changed)
-		return Handle(createLink(t, oset_results, expr->getTruthValue()));
+	{
+		LinkPtr subl = createLink(t, oset_results, expr->getTruthValue());
+		return Handle(_as->add_atom(subl));
```

Comment:
This wasn't done before, and should not be  needed now, because the top-most result is added to the atomspace; the lower-level results do not need to be.

Kind-of off-topic, but I should mention this: the instantiator, as currently written, is a bit mis-designed; it does eager evaluation, and should really do lazy evaluation instead.